src/runtime/slot-composer.ts

```diff
@@ -169,6 +169,10 @@ export class SlotComposer {
     const slotConsumer = this.getSlotConsumer(particle, slotName);
     assert(slotConsumer, `Cannot find slot (or hosted slot) ${slotName} for particle ${particle.name}`);
 
+    // Content object as received by the particle execution host is frozen.
+    // SlotComposer attach properties to this object, so we need to clone it at the top level.
+    content = {...content};
```

Comment:
slot-composer is in process of being deprecated.
ui-slot-composer is the new one that's being used. so probably it should be updated too. but you'd want to coordinate it with @sjmiles because he'll be fixing the circular dependency thing now.

Reply:
My thinking is:
1) SlotComposer is being deprecated, so I'm not too worried about putting in slightly questionable code.
2) UiSlotComposer is being fixed, so I'm leaving it to SJM to bring the required changes.

Does this make sense?